### PR TITLE
Use the existing and configured env indicator

### LINKED ISSUE #475: Replace PYTHON_ENV with ENVIRONMENT
## Description

Our deployment infrastructure already configures an `ENVIRONMENT` variable and it is already used in the ingestion server code. Rather than introduce a new one that duplicates that, we can just use that everywhere.

Making this change will solve the issue we're seeing Sentry where staging is reporting itself as `production`.

## Resolution

- [x] 🙋 I would be interested in resolving this bug.

### PR DESCRIPTION
**Pull request by sarayourfriend**
## Fixes

Fixes #475 by @sarayourfriend 

## Description

Replace `PYTHON_ENV` with the already used `ENVIRONMENT`.

## Testing Instructions

Checkout the branch and update your local instance (`just recreate && just init` should do it but will wipe local data) and verify the API runs.

Unit tests should also pass of course.

## Checklist

- [x] My pull request has a descriptive title (not a vague title like `Update index.md`).
- [x] My pull request targets the *default* branch of the repository (`main`) or a parent feature branch.
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [N/A] I added or updated tests for the changes I made (if applicable).
- [N/A] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no visible errors.

[best_practices]:https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

File: api/catalog/settings.py [hunk 1]
```diff
@@ -39,7 +39,7 @@
 # SECURITY WARNING: don't run with debug turned on in production!
 DEBUG = config("DJANGO_DEBUG_ENABLED", default=False, cast=bool)
 
-PYTHON_ENV = config("PYTHON_ENV", default="production")
+ENVIRONMENT = config("ENVIRONMENT", default="local")
```
Comment (krysal) [inline api/catalog/settings.py:42]: Why did it change to "local"?

Comment (sarayourfriend) [reply api/catalog/settings.py:42]: It's the default that is used by the ingestion_server when it accesses this same variable:

https://github.com/WordPress/openverse-api/blob/ad2f74a111af4ea35f939ec597f394c83f6dbc23/ingestion_server/ingestion_server/distributed_reindex_scheduler.py#L73

Comment (krysal) [reply api/catalog/settings.py:42]: I think we should keep "production" and change to "staging" or the corresponding string in the staging server. "local" doesn't seem very useful as everything is local to its own machine.

Comment (AetherUnbound) [reply api/catalog/settings.py:42]: Yea I agree that naming convention isn't too useful! :sweat_smile: I think `dev` or `staging` makes sense. A lot of the environment checks in the ingestion server are checking explicitly if in prod (`environment == "prod"`) so it hasn't mattered much what _not prod_ is.

**Review by AetherUnbound — APPROVED** [review]
Looks good! We can follow up with a future PR to change "local" to something clearer :slightly_smiling_face:

**Review by krysal — APPROVED** [review]

File: api/catalog/settings.py (hunk 1 above)
Comment (sarayourfriend) [reply api/catalog/settings.py:42]: I agree `local` is ambiguous.

In my mind (and indeed I've seen this in practice) `dev` is local development, `staging` is the pre-production deployment, `prod` is production, and `test` is when tests are running regardless of environment. Also variations on those strings exist. Node (and most JavaScript projects like React and Vue) uses the full `development` and `production` specifically for example, so I like the consistency of using that for the less-opinionated Python environment.

**Pull request merged**